nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/factory/Nd4j.java

```diff
@@ -5615,7 +5657,22 @@ public static void writeAsNumpy(INDArray arr, OutputStream writeTo) throws IOExc
             }
 
             bufferedOutputStream.flush();
+            return written;
         }
+
+    }

```

Comment:
That looks quite repetitive, no way to rewrite this such that the `closeFlush` flag doesn't produce duplicate code?